### PR TITLE
Add default serialization config to apps

### LINKED ISSUE #235: Remove avro dependency

### PR DESCRIPTION
**Pull request by philipp94831**
Fixes #235

### CODE REVIEW

**Comment by philipp94831** [comment]
@torbsto kafka-schema-registry-client still brings Avro so this is the best we can do

**Review by torbsto — APPROVED** [review]

**Review by raminqaf — APPROVED** [review]

File: streams-bootstrap-core/src/main/java/com/bakdata/kafka/ProducerApp.java [hunk 1]
```diff
@@ -46,4 +45,7 @@ default ProducerCleanUpConfiguration setupCleanUp(
             final EffectiveAppConfiguration<ProducerTopicConfig> configuration) {
         return new ProducerCleanUpConfiguration();
     }
+
+    @Override
+    SerializerConfig defaultSerializationConfig();
```
Comment (JakobEdding) [inline streams-bootstrap-core/src/main/java/com/bakdata/kafka/ProducerApp.java:50]: Can you add a javadoc? Also for StreamsApp.

Comment (philipp94831) [reply streams-bootstrap-core/src/main/java/com/bakdata/kafka/ProducerApp.java:50]: It is inherited from [App](https://github.com/bakdata/streams-bootstrap/pull/239/files#diff-08a4a32d60f8228190e670de077f3b54219d31c25fb8b4510e063f432c9a35cfR70)

Comment (philipp94831) [reply streams-bootstrap-core/src/main/java/com/bakdata/kafka/ProducerApp.java:50]: I just enforce return type here to be SerializerConfig and not the interface

**Review by JakobEdding — APPROVED** [review]

**Review by raminqaf — APPROVED** [review]

**Pull request merged**